packages/grid/_modules_/grid/models/colDef/gridDateOperators.ts

```diff
@@ -2,24 +2,45 @@ import { GridFilterInputValue } from '../../components/panel/filterPanel/GridFil
 import { GridFilterItem } from '../gridFilterItem';
 import { GridFilterOperator } from '../gridFilterOperator';
 
+const dateRegex = /(\d+)-(\d+)-(\d+)/;
+const dateTimeRegex = /(\d+)-(\d+)-(\d+)T(\d+):(\d+)/;
+
+function buildApplyFilterFn(
+  valueToFilter: string,
+  compareFn: (value1: number, value2: number) => boolean,
+  showTime?: boolean,
+) {
+  const [year, month, day, hour, minute] = valueToFilter
+    .match(showTime ? dateTimeRegex : dateRegex)!
+    .slice(1)
+    .map(Number);
+
+  const time = new Date(year, month - 1, day, hour || 0, minute || 0).getTime();
+
+  return ({ value }): boolean => {
+    if (!value) {
+      return false;
+    }
+    // Make a copy of the date to not reset the hours in the original object
+    const valueAsDate = new Date(value instanceof Date ? value : value.toString());
+    const timeToCompare = valueAsDate.setHours(
+      showTime ? valueAsDate.getHours() : 0,
+      showTime ? valueAsDate.getMinutes() : 0,
+      0,
+      0,
+    );
+    return compareFn(timeToCompare, time);
```

Comment:
Should it have?

```suggestion
    return compareFn(timeToCompare.getTime(), time);
```

Reply:
`timeToCompare` is already the number of milliseconds.

Reply:
Oh wow, ok, we rely on the return value of setHours, smart.